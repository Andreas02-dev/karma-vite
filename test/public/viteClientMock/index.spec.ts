/**
 * @jest-environment jsdom
 * @jest-environment-options {"url": "https://jestjs.io/"}
 */
const envMock = jest.fn();
jest.mock(
  '@vite/env',
  () => {
    envMock();
  },
  { virtual: true },
);

import * as output from '@/public/viteClientMock';

describe('viteClientMock', () => {
  it('export correctly', () => {
    expect(output).toMatchSnapshot();
  });

  it('imported @vite/env', () => {
    jest.resetModules();
    require('@/public/viteClientMock');

    expect(envMock).toHaveBeenCalled();
  });
});
