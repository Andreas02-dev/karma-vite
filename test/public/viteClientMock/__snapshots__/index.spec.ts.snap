// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viteClientMock export correctly 1`] = `
{
  "ErrorOverlay": [Function],
  "createHotContext": [Function],
  "injectQuery": [Function],
  "removeStyle": [Function],
  "updateStyle": [Function],
}
`;
